single_headers/lithium_http_server.hh

```diff
@@ -4056,12 +4056,20 @@ struct output_buffer {
   }
 
   output_buffer& operator<<(std::string_view s) {
-    if (cursor_ + s.size() >= end_)
-      flush();
-
-    assert(cursor_ + s.size() < end_);
-    memcpy(cursor_, s.data(), s.size());
-    cursor_ += s.size();
+    size_t pos = 0;
+    while (pos < s.size()) {
+      size_t space = end_ - cursor_;
```

Comment:
`size_t` is unsigned which means a negative value from `end_ - cursor_` could get access to memory out of range.